% ******************************* Thesis Appendix C ********************************

\chapter{Additional Material for Chapter 4}

\graphicspath{{Appendix3/Figs/}}

\section{Experimental methods}

\subsection{Experimental mice and infections}

Wild-type and transgenic inbred mouse strains were housed and used in blood-stage Plasmodium infections, as described in Supplementary Materials and Methods.

\subsection{Flow cytometry}

Splenocytes were isolated and assessed by flow cytometry as described in Supplementary Materials and Methods.

\subsection{Single-cell mRNA sequencing}

Single-cell capture and processing, as well as quality control analysis of scRNA-seq data, were performed as described in Supplementary Materials and Methods.

\subsection{Statistics}

Statistical analyses were conducted using R, Python, or GraphPad Prism. The types of statistical tests and significance levels are described in respective figure legends.

\subsection{Experimental mice, adoptive transfer and infections}

\begin{sloppypar}
C57BL/6, rag1-/-, and congenic PTprca mice were purchased from Australian Resource Center (Canning Vale) or bred in-house. PbTII, C57BL/6, rag1-/-, congenic PTprca (CD45.1), nzEGFP, lgals1-/- (Jackson Laboratory: Stock No: 006337), LysMCre (Jackson Laboratory Stock No: 004781), ROSA26iDTR (iDTR) (Jackson Laboratory Stock No: 007900) mice, and all crosses were maintained under specific pathogen-free conditions within animal facilities at the Wellcome Trust Genome Campus Research Support Facility (Cambridge, UK), registered with the UK Home Office, or at QIMR Berghofer Medical Research Institute (Brisbane, Australia). All mice were female and used at 8-12 weeks of age. All animal procedures were in accordance with the Animals (Scientific Procedures) Act 1986 and approved by the Animal Welfare and Ethical Review Body of the Wellcome Trust Genome Campus, or in accordance with Australian National Health and Medical Research Council guidelines and approved by the QIMR Berghofer Medical Research Institute Animal Ethics Committee (approval no. A02-633M).
\end{sloppypar}

Spleens from PbTII donor mice were aseptically removed and homogenized through a 100 \( \mu \)m strainer before erythrocytes lysis using RBC lysis buffer (eBioscience). CD4+ T cells were enriched using CD4 microbeads (Miltenyi Biotech) and stained with CellTrace™ Violet (Invitrogen). Cells were injected (106/200\( \mu \)l RPMI) via a lateral tail vein.

PcAS parasites were used after one in vivo passage in WT C57BL/6 mice. Mice were infected with 105 pRBCs i.v. and blood parasitemia was monitored by Giemsa-stained thin blood smears obtained from tail bleeds.

\subsection{Flow Cytometry}

Single-cell suspensions were prepared by homogenizing spleens through 100 \( \mu \)m strainers and lysing erythrocytes using RBC lysis buffer (eBioscience). Prior to staining, Fc receptors were blocked using anti-CD16/32 antibody (BD Pharmingen or in-house). Intracellular staining was performed by first incubating cells in brefeldin-A (10 mg/ml) at 37oC for 3 hours. For IL-10/IFN\( \gamma \) staining, cells were also incubated with Ionomycin (500 ng/ml) and PMA (25 ng/ml). Staining was performed using the eBioscience FoxP3 intracellular kit. For DNA/RNA staining, Hoechst33342 (10\( \mu \)g/ml; Sigma) was added at 1/500 v/v to cell preparation 15 minutes prior to acquisition using a BD LSRFortessa IV (BD Bioscience). Cells were sorted using a MoFlo XDP (Beckman Coulter), a FACSAria II (Becton Dickinson) or an Influx (Becton Dickinson) instrument. Activated PbTII cells were sorted as CD4+TCR\( \beta \)+ and CD69+ and/or divided at least once as measured using the CellTraceTM Violet proliferation dye. Dendritic cells were sorted as CD11chiMHCIIhiTCR\( \beta \)-B220-. Naive dendritic cells were further sorted as CD8\( \alpha \)+CD11b- or CD8\( \alpha \)-CD11b+, and inflammatory monocytes as CD11bhiLy6ChiLy6GloTCR\( \beta \)-B220-.

\subsection{Single-cell capture and processing}

Single cell processing with the Fluidigm C1 system was performed using small–sized capture chips (for 5-10 \( \mu \)m cells). 1 \( \beta \)l of a 1:4000 dilution of External RNA Control Consortium (ERCC) spikeins (Ambion, Life Technologies) was included in the lysis buffer. For processing with the Smartseq2 protocol, the cells were sorted into 96-well plates containing lysis buffer. The lysis buffer consisted of Triton-X, RNase inhibitor, dNTPs, dT30 primer and ERCC spike-ins (Ambion, Life Technologies, final dilution 1:100 million). 24 cycles of cDNA amplification were performed. Libraries were prepared using Nextera XT DNA Sample Preparation Kit (Illumina), pooling up to 96 single cells. Pooled libraries were purified using AMPure XP beads (Beckman Coulter) and sequenced on an Illumina HiSeq 2500 instrument, using paired-end 100 or 125-base pair reads.

\subsection{Processing and QC of scRNA-Seq data}

Gene expression was quantified using Salmon, version 0.4.0. The parameter libType=IU, and a transcriptome index built on Ensembl version 78 mouse cDNA sequences. Sequences from the ERCC RNA spike-ins were included in the index, as well as 313 mouse-specific repeat sequences from RepBase. As quality control measures, we assessed the number of input read pairs, and the amount of mitochondrial gene content, considering cells with less than 100,000 reads or more than 35\% mitochondrial gene content as failed. For T cells, we additionally considered cells where number of genes was less than 100 + 0.008 * (mitochondrial gene content) as failed. For the data generated using a 96-well plate-based Smart-seq2 protocol, which does not permit visual inspection of the captured cells, we additionally excluded low-quality cells from which fewer than 2000 genes were detected, motivated by negative control wells. To verify that that the cells sorted in the wells were PbTII cells, we only selected cells from which both the transgenic TCR alpha and beta chains were detected (Supplementary Tables 2 and 3). For expression values, the Transcripts Per Millions (TPM's) estimated by Salmon included ERCC spike-ins. Thus, to obtain values representing only the endogenous RNAs, we removed ERCC's from the expression table and scaled the TPM's so they again summed to a million. We also globally removed genes from analysis where less than three cells expressed the gene at minimum 1 TPM, unless stated otherwise.

\subsection{Determining T cell receptor expression}

T cell receptor sequences were reconstructed from scRNAseq data using the TraCeR software as
previously described \cite{Stubbington2016-dt}.

\subsection{Annotation of cell-surface receptors, cytokines and transcription factors}

\begin{sloppypar}
Genes likely to encode transcription factors, cell-surface receptors or cytokines were found by combining information from KEGG (http://www.genome.jp/kegg/), the Gene Ontology Consortium (http://geneontology.org/, PANTHER (http://www.pantherdb.org/) along with the more specific databases detailed below.
\end{sloppypar}

\begin{sloppypar}
Transcription factors were found by searching the Gene Ontology Consortium database using the following ontology term: GO:0003700 (sequence-specific DNA binding transcription factor activity); KEGG for ko03000 (Transcription Factors); PANTHER for PC00009 (DNA binding) AND PC00218 (Transcription Factors). The presence of genes in the following databases was also used as evidence for transcription factor activity: AnimalTFDB (http://www.bioguo.org/AnimalTFDB/index.php), DBD (http://www.transcriptionfactor.org), TFCat (http://www.tfcat.ca), TFClass (http://tfclass.bioinf.med.uni-goettingen.de/tfclass), UniProbe (http://the\_brain.bwh.harvard.edu/uniprobe) and TFcheckpoint (http://www.tfcheckpoint.org).
\end{sloppypar}

Cell-surface receptors were found by searching the Gene Ontology Consortium database using the following ontology terms GO:0004888 (transmembrane signaling receptor activity) OR GO:0008305 (integrin complex)) AND NOT (GO:0004984 (olfactory receptor activity) OR GO:0008527 (taste receptor activity); KEGG for ko04030 (G-Protein Coupled Receptors) OR 64 ko04050 (Cytokine Receptors) OR ko01020 (Enzyme-linked Receptors); PANTHER for PC00021 (G-Protein Coupled Receptors) OR PC00084 (Cytokine Receptors) OR PC00194 (Enzyme-linked Receptors). Annotation of genes as receptors in the ImmPort (https://immport.niaid.nih.gov/), GPCRDB (http://gpcrdb.org/) or IUPHAR (http://www.guidetopharmacology.org/) databases was also used as evidence for receptor functionality.

Cytokines were found by searching the Gene Ontology Consortium database using the following ontology terms GO:0005125 (cytokine activity); KEGG for ko04052 (Cytokines); PANTHER for PC00083 (Cytokines). Annotation of genes as cytokines in ImmPort was also used in this case. Genes were scored according to the number of databases and search results in which they occurred. Scores were weighted according to the strength of evidence provided by each database such that functional annotations supported by manually reviewed experimental evidence were given a higher score than those that were solely computationally generated (Table)

Genes were assigned as likely cell-surface receptors or cytokines if they had a cumulative score greater than or equal to 5 in that category. Genes were assigned as likely transcription factors if they had a cumulative score greater than or equal to 6 in that category.

\subsection{\textit{In vivo} depletion}

Cellular depletion in LysMCre x iDTR mice was performed by intraperitoneal injection of 10ng/g DT (Sigma-Aldrich) in 200\( \mu \)l 0.9\% saline (Baxter) at day 3 post-infection. Control mice were given 0.9\% saline only. For B cell depletion, anti-CD20 (Genentech) or isotype control antibody was administered in a single 0.25mg dose via i.p. injection in 200\( \mu \)l 0.9\% NaCl (Baxter), 7 days prior to infection.

\subsection{Confocal microscopy}

Confocal microscopy was performed on 10–20 \( \mu \)m frozen spleen sections. Briefly, splenic tissues were snap frozen in embedding optimal cutting temperature (OCT) medium (Sakura) and stored at -80oC until use. Sections were fixed in ice-cold acetone for 10 minutes prior to labeling with antibodies. DAPI was used to aid visualization of white pulp areas. Samples were imaged on a Zeiss 780-NLO laser-scanning confocal microscope (Carl Zeiss Microimaging) and data analyzed using Imaris image analysis software, version 8.1.2 (Bitplane). Cells were identified using the spots function in Imaris, with thresholds <10mM and intensities <150. All objects were manually inspected for accuracy before data were plotted and analyzed in GraphPad prism (version 6).

\section{Additional figures}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{"Fig S1 rev3"}
    \caption[Enrichment of PbTII cells for adoptive transfer]{\textbf{Enrichment of PbTII cells for adoptive transfer.} (A) CD4+ T cells were enriched using positive selection (MACS microbeads) from the spleen of a naive, PbTII x CD45.1 mouse. FACS plots show purity, expression of V\( \alpha \)2 and V\( \beta \)12 transgenes, and CellTrace™ Violet (CTV) staining of enriched PbTII cells compared to corresponding flowthrough from the enrichment process.}
    \label{fig:ms1}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{"Fig S2 rev3"}
    \caption[Sorting strategy for PbTII cells]{\textbf{Sorting strategy for PbTII cells.} (A) PbTII cells (CD4+ TCR\(\beta\)+ CD45.1+) were adoptively transferred into WT congenic (CD45.2+) recipient mice At indicated days, early activated (CD69+) and/or proliferated (CTVlo) PbTII cells were cell-sorted from spleens of PcAS-infected mice, and naive PbTII cells (CD69loCTVhi) were cell-sorted from the spleens of naive mice at day 7 post-transfer.}
    \label{fig:ms2}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{"Fig S3 rev3"}
    \caption[Flow cytometric assessment of T 1/T responses during PcAS infection]{\textbf{Flow cytometric assessment of T 1/T responses during PcAS infection.} (A) Flow cytometic gating strategy employed to analyze splenic PbTII responses throughout this manuscript. (B) Isotype controls for direct ex vivo intracellular staining of IFN\( \gamma \), T-bet and Bcl6, and fluorescence minus one (FMO) control for staining of CXCR5 expression by splenic PbTII cells from day 7-infected mice. (C) Remaining FACS plots from data in Fig. 1B-C, showing expression of T-bet and Bcl6 by IFN\( \gamma \)+ or CXCR5+ splenic PbTII cells at day 7 post-infection with PcAS. Each plot represents an individual mouse.}
    \label{fig:ms3}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.8\textwidth]{"Fig S4 rev3"}
    \caption[Expression of subset-specific marker genes in PbTII cells]{\textbf{Expression of subset-specific marker genes in PbTII cells.} (A) Representative FACS plot (gated on CD4+ TCR\( \beta \)+ live singlets) and proportion of FOXP3+ (Treg) splenic PbTII (104 transferred) (CD45.1+; red dashed box) or polyclonal CD4+ T (CD45.1-; black dashed box) cells from mice (n=6) at day 7 post-infection. (B-C) FACS plots (gated on CD45.1+ CD4+ TCR\( \beta \)+ live singlets) of (B) IL-4+GATA3+ (Th2) and (C) IL-17+ROR\( \gamma \)t+ (Th17) splenic PbTII cells in naive (receiving 106 cells) or PcAS-infected mice (receiving 104 cells) at day 7 post-infection. (A-C) Data are representative of two independent experiments. Statistics: Mann-Whitney U test; *p<0.05. (D) The mRNA expression of selected subset-specific cytokines and the Treg hallmark transcription factor Foxp3 in PbTII cells. The red dots and line indicate the fraction of cells in each time point where the particular mRNA was detected.}
    \label{fig:ms4}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.8\textwidth]{"Fig S5 rev3"}
    \caption[Heterogeneity of activated PbTII cells and variability associated with cell size and differentiation]{\textbf{Heterogeneity of activated PbTII cells and variability associated with cell size and differentiation.} (A) PCA of single PbTII cells at 2, 3, 4 and 7 days post-infection with PcAS. The PCA was based on all genes expressed at \( \geq 100 \) TPM in at least 2 cells. The arrows represent the Pearson correlation with PC1 and PC2. Cell size refers to the number of detected genes. “Th1 signature” and “Tfh signature” refer to cumulative expression of top 30 signature genes associated with Th1 and Tfh phenotypes (15). The numbers in parenthesis show proportional contribution of respective PC. (B) The relationship of detected cell number with the fraction or reads mapping to ERCC spikein RNA (top) and with cumulative expression of proliferation markers Mki67, Mybl2, Bub1, Plk1, Ccne1, Ccnd1 and Ccnb1 (31) (Figure 4B and S9). (C) Ranked loading scores for PC1-PC6 of the Th1 and Tfh signature genes in the PCA shown in (A). The numbers in parenthesis show proportional contribution of respective PC.}
    \label{fig:ms5}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.8\textwidth]{"Fig S6 rev3"}
    \caption[Heterogeneity of T 1/T H signature gene expression in activated PbTII cells]{\textbf{Heterogeneity of T 1/T H signature gene expression in activated PbTII cells.} (A) Principal component analyses of day 4 (left) and day 7 (right) PbTII cells were performed using established Th1/Tfh signature genes (15) detected at the level \( \geq 100 \) TPM in at least 2 cells. The numbers in parenthesis show proportional contribution of respective PC. (B) The PC1 and PC2 loadings of individual Th1 (red) and Tfh (blue) signature genes in PCA of day 4 and day 7 PbTII cells (A). (C) The correlation of PC1 from the analysis with the signature genes alone and PC2 of the genome-wide analysis (Figure 1E).}
    \label{fig:ms6}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.8\textwidth]{"Fig S7 rev3"}
    \caption[Heterogeneity of the entire PbTII time series and the contribution of T 1 and T genes to the overall variability]{\textbf{Heterogeneity of the entire PbTII time series and the contribution of T 1 and T genes to the overall variability.} (A) The first five components of the Principal Component Analysis of the entire time series. The numbers in parenthesis show proportional contribution of respective PC. (B) The rankings of the Th1 and Tfh signature genes among the loadings of Principal Components 1-7.}
    \label{fig:ms7}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.8\textwidth]{"Fig S8 rev3"}
    \caption[The relationship of pseudotime with time points and with the T 1 assignment probability]{\textbf{The relationship of pseudotime with time points and with the T 1 assignment probability.}}
    \label{fig:ms8}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{"Fig S9 rev3"}
    \caption[Correlation of GPfates trends with TH1 and T H signature genes]{\textbf{Correlation of GPfates trends with TH1 and T H signature genes.} (A) The effect of the probability threshold on the cumulative expression of TH1 and T H signature genes (15). The p-values were calculated using Wilcoxon rank sum test. (B) Correlation of expression of T 1 and T 1 assignment probability. (C) Relation between genes expression correlation with mixture assignment probability, and the bifurcation statistic, for each gene. The threshold of bifurcation statistic = 49 has some stronger effect sizes. This is analogous to a volcano plot in classical differential expression testing.}
    \label{fig:ms9}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{"Fig S10 rev3"}
    \caption[Expression of transgenic and endogenous TCRs]{\textbf{Expression of transgenic and endogenous TCRs.} (A) Statistics of TCR sequence detection. Numbers correspond to single cells in which the corresponding transcript was detected. (B) Expression levels (log2(TPM)) of for the endogenous or transgenic TCR\( \alpha \) chains across the entire dataset.}
    \label{fig:ms10}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{"Fig S11 rev3"}
    \caption[Expression of endogenous TCRs does not influence PbTII cell T 1/TFH differentiation]{\textbf{Expression of endogenous TCRs does not influence PbTII cell T 1/TFH differentiation.} (A) Representative FACS plots (gated on CD45.1+ (WT) or CD45.2+ (Rag1-/-), CD4+ TCR\( \beta \)+ V\( \beta \)12+ live singlets) showing expression of T-bet or Bcl6 by splenic WT or Rag1-/- PbTII cells (104 transferred into congenic recipient mice) at day 7 p.i. with PcAS (n=4). (B) Summary graphs of proportions of WT or Rag1-/- PbTII cells exhibiting T-bethi and Bcl6hi phenotypes from (A). (C) Representative histograms of CXCR5 and IFN\( \gamma \) expression by T-bethi or Bcl6hi WT and Rag1-/- PbTII cells from (A) \& (B). Statistics: Mann-Whitney U test; NS, not significant.}
    \label{fig:ms11}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.7\textwidth]{"Fig S12 rev3"}
    \caption[Robustness of top bifurcating genes across experiments]{\textbf{Robustness of top bifurcating genes across experiments.} (A) Experimental design for the replicate PcAS infection. Single cells were sorted into 96-well plates and cDNA was amplified using the Smart-seq2 protocol. (B) Bayesian Gaussian Process Latent Variable model of the combination of original and replicate data. The BGPLVM was fitted using the residuals from an ordinary least squares model of expression from the categorical variable of experiment, equivalent to limma::removeBatchEffect. Replicate data groups with corresponding data from the original experiment, illustrating that both experiments capture the same transcriptional landscape. (C) The emergence of subset-specific gene patterns at day 7 of infection. For the top bifurcating genes (Fig S5C) pairwise gene-to-gene Spearman correlations were calculated. The rowside colours represent the association of the gene with either Th1 fate (red) or Tfh fate (blue). (D) The expression of top 20 Th1 and Tfh associated genes identified using GPfates in single PbTII cells at days 4 and 7. The genes were annotated as Th1- or Tfh-associated based on public datasets (15, 37, 4 , ). *Cdk2ap2 appears twice because two alternative genomic 4 47 annotations exist.}
    \label{fig:ms12}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.8\textwidth]{"Fig S13 rev3"}
    \caption[Flow cytometric validation of CXCR6 expression in PbTII cells prior to and after bifurcation]{\textbf{Flow cytometric validation of CXCR6 expression in PbTII cells prior to and after bifurcation.} (A) Representative FACS plots showing kinetics of CellTraceTM Violet (CTV) dilution and CXCR6 expression, with summary graphs showing proportion of PbTII cells expressing this (after 106 PbTII cells transferred) in un-infected (Day 0) and PcAS-infected mice at indicated days postinfection (n=4 mice/time point, with individual mouse data shown in summary graphs; solid line in summary graphs indicates results from third order polynominal regression analysis.) Data are representative of two independent experiments. (B) Representative FACS plots showing CXCR6 expression in Tbethi (red gate) and Bcl6hi (blue gate) PbTII cells, compared to naive PbTIIs (gray) at 7 days post-infection. Summary graph shows mean \& standard deviations for geometric mean fluorescence intensity of CXCR6 expression in gated PbTII populations (n=4 mice) Statistics: Mann-Whitney U test *p<0.05.}
    \label{fig:ms13}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{"Fig S14 rev3"}
    \caption[T cell-intrinsic Galectin-1 supports T 1 fate commitment]{\textbf{T cell-intrinsic Galectin-1 supports T 1 fate commitment.} (A) Expression of Lgals1 in the GPfates model across pseudotime. Curves represent Th1 (red) and Tfh (blue) trends when weighing the information from data points according to trend assignment. (B) Histograms of Galectin-1 expression by splenic PbTII cells (n=3-6 mice per group, all data shown overlaid within each groups) and proportions expressing Galectin-1 in naive mice (106 transferred; gray), and by T 1 (T-bethi IFN\( \gamma \)+; blue) and T H (Bcl6 h iCXCR5 +; green) cells (10 4 transferred) in PcAS-infected mice at day 7 post-infection. Statistics: Mann-Whitney U Test; ** p<0.01. Data are representative of two independent experiments. (C) Schematic showing co-transfer of WT (GFP+ CD45.2+) and Lgals1-/- (CD45.2+) PbTII cells (104 of each transferred) into WT congenic CD45.1+ recipient mice (n=10), and gating strategy for assessment of splenic PbTII cells at 7 days post-infection. (D) Representative FACS plots (gated on GFP+ or GFP-, CD45.2+ CD4+ TCR\( \beta \)+ V\( \beta \)12+ live singlets) and paired analysis of proportions of splenic WT and Lgals1-/- PbTII cells exhibiting Tbethi IFN\( \gamma \)+ (T 1) and Bcl6hi CXCR5+ (T Wilcoxon signed-rank Pairwise T-test; **p<0.01; NS, not significant.}
    \label{fig:ms14}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.66\textwidth]{"Fig S15 rev3"}
    \caption[IL-10- and IFN\( \gamma \)-coproducing Tr1 cells derive from T 1 cells]{\textbf{IL-10- and IFN\( \gamma \)-coproducing Tr1 cells derive from T 1 cells.}(A) The expression kinetics of Ifng (left) and Il10 (right) according to the GPfates model. Curves represent the expression patterns associated with the T 1 (red) and the T (B) Co-expression of Ifng and Il10 in single cells. The colors of the data points represent time points and the shapes represent cells from two replicate experiments. Tr1 cells were defined as cells expressing both Ifng and Il10 at \( \geq 10 \) TPM. T 1 cells were defined as cells expressing Ifng but not Il10 at \( \geq 10 \) TPM. (C) Representative FACS plots (gated on CD45.1+ CD4+ TCR\( \beta \)+ live singlets), proportions and mean fluorescence intensities of IFN\( \gamma \) (T 1) and IL-10+ IFN\( \gamma \)+ (Tr1) PbTII cells (104 transferred) with or without ex vivo PMA/ionomycin restimulation at day 7 post-infection with PcAS. Statistics: Mann-Whitney U test; *p<0.05. Geom Mean FL; Geometric Mean Fluorescence Level. (D) Differential expression genes between day 7 T 1 cells and Tr1 cells, as defined in (B). All genes expressed in at least 20\% of the single cells were included in the analysis. P-values were calculated using Wilcoxon Rank Sum test, and adjusted for multiple testing using Benjamini \& Hochberg correction. The top hit Il10 is not shown. (E) Analysis of expression frequency for all genes in the day 7 T 1 cells and Tr1 cells, as defined in (B). Expression frequency was defined as the number of cells where the transcript was detected, HFH (blue) trends. divided by total number of cells. Genes with at least 0.3 difference in expression frequency between T 1 and Tr1 cells are highlighted in red.}
    \label{fig:ms15}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.8\textwidth]{"Fig S16 rev3"}
    \caption[Proliferative burst of activated PbTII cells]{\textbf{Proliferative burst of activated PbTII cells.} (A) Fluorescence minus one (FMO) control for expression of Ki67 by splenic PbTII cells from a day 7-infected mouse. (B) The expression of established proliferation genes (31) along pseudotime. (C) ModFit plots and proportions of PbTII cells in G0/G1, G2/M and S-phase of cell cycle as determined by Hoechst staining.}
    \label{fig:ms16}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{"Fig S17 rev3"}
    \caption[Kinetics of chemokine receptor expression during PcAS infection according to the GPfates model]{\textbf{Kinetics of chemokine receptor expression during PcAS infection according to the GPfates model.} Curves represent the expression patterns associated with the T 1 (red) and the T (blue) trends.}
    \label{fig:ms17}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.8\textwidth]{"Fig S18 rev3"}
    \caption[Coexpression of chemokine receptors at single-cell level during PcAS infection]{\textbf{Coexpression of chemokine receptors at single-cell level during PcAS infection.} (A) The expression of chemokine receptors in single cells at day 4 post infection. (B) The expression of chemokine receptors in single cells at day 7 post infection. (C) Representative FACS plots and proportions of splenic PbTII cells co-expressing CXCR5 and CXCR3 in naive (gray; n=3) or infected mice (green; n=6) at 4 days post-infection with PcAS. Results are representative of two independent experiments. Statistics: Mann-Whitney U test *p<0.05.}
    \label{fig:ms18}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{"Fig S19 rev3"}
    \caption[B cells are essential for T responses in PbTII cells during PcAS infection]{\textbf{B cells are essential for T responses in PbTII cells during PcAS infection.} Representative FACS plots (gated on CD4+ TCR\( \beta \)+ CD45.1+ live singlets) of splenic PbTII cells, showing proportions exhibiting T H (Bcl6+ CXCR5+) and T \( \gamma \)+) phenotypes in WT mice (receiving 104 PbTII cells), treated with anti-CD20 monoclonal antibodies (0.25mg) to deplete B-cells, or control IgG, and infected for 7 days with PcAS. Individual mice data (n=5) shown in summary graph. Mann-Whitney U test *p<0.05; **p<0.01. Results are representative of two independent experiments.}
    \label{fig:ms19}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{"Fig S20 rev3"}
    \caption[Sorting strategy for myeloid cells]{\textbf{Sorting strategy for myeloid cells.} Representative FACS plots showing sorting strategy for CD8\( \alpha \)+ and CD11b+ cDC, and Ly6Chi inflammatory monocytes from the spleens of naive and 3-day PcAS-infected mice.}
    \label{fig:ms20}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{"Fig S21 rev3"}
    \caption[PCA of cDCs from naive and infected mice]{\textbf{PCA of cDCs from naive and infected mice.} Results of Principal Component (PC) Analysis on scRNA-seq mRNA reads (filtered by minimum expression of 100 TPM in at least 2 cells) from 131 single splenic naive CD8\( \alpha \)+ and CD8\( \alpha \)- and mixed day 3 PcAS-infected cDC. PC1-PC6 shown. Axis labels show proportional contribution of respective PC.}
    \label{fig:ms21}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.8\textwidth]{"Fig S22 rev3"}
    \caption[Differential gene expression between single splenic CD8\( \alpha \)+ and CD8\( \alpha \)- cDCs]{\textbf{Differential gene expression between single splenic CD8\( \alpha \)+ and CD8\( \alpha \)- cDCs.} (A) Results of differential gene expression analysis between naive splenic CD8\( \alpha \)+ and CD8\( \alpha \)- cDCs, for all genes expressed in greater than 2 cells. (B) Complete list of differentially-expressed genes between naive CD8\( \alpha \)+ and CD8\( \alpha \)- cDCs, which were expressed in >10 cells of either subset with a qval <0.2 as determined in (A). (C) Heatmap of naive cDCs ordered by PC2 (Fig. 6A) and expression of genes from (B) ordered by PC2 loading in (Fig 6A). (D) Heatmap examining hierarchical clustering of mixed CD8\( \alpha \)+ and CD8\( \alpha \)- CD11b+ day 3- infected cDCs (cell-sorted and mixed at a ratio of 50:50 prior to scRNA-seq) using differentially expressed genes from (B) ordered by PC2 loading shown in (Fig 6A).}
    \label{fig:ms22}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.1\textwidth]{"Fig S23 rev3"}
    \caption[Differentially expressed genes between single naive and day 3 PcAS-infected cDCs]{\textbf{Differentially expressed genes between single naive and day 3 PcAS-infected cDCs.} List of differentially expressed genes, expressed in >10 cells (qval<0.05) between naive and day 3- infected cDCs. Mean TPM fold-change in gene expression relative to naive levels.}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{"Fig S24 rev3"}
    \caption[PCA of Ly6Chi monocytes from naive and infected mice]{\textbf{PCA of Ly6Chi monocytes from naive and infected mice.} Results of Principal Component (PC) Analysis using scRNA-seq mRNA reads (filtered by minimum expression of 100 TPM in at least 2 cells) of 154 single splenic Ly6Chi monocytes from naive and infected mice. PC1-PC6 shown. Axis labels show proportional contribution of respective PC.}
    \label{fig:ms24}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.8\textwidth]{"Fig S27 rev3"}
    \caption[Myeloid cell depletion in LysMCre x iDTR mice]{\textbf{Myeloid cell depletion in LysMCre x iDTR mice.} LysMCre x iDTR mice were infected with PcAS, and treated 3 days later with DT (10ng/g intraperitoneal injection) or control saline (n=6 per group). 24 hours later spleens were harvested for cellular compositional analysis: (A) Representative FACS plots enumerating splenic inflammatory monocytes (Ly6Chi CD11bhi Ly6G- B220- TCR\( \beta \)-). (B) Representative fluorescence micrographs showing spleen tissue sections co-stained for B cells (B220 in red) and macrophages (CD68 (top panel) or SIGN-R1 (bottom panel) in green) and summary graphs of average cell number in three fields of view covering the total cross section of a spleen. (C) Flow cytometric enumeration of splenic cDC (CD11chi MHCIIhi B220- TCR\( \beta \)-).}
    \label{fig:ms27}
\end{figure}

